workspaces/src/rpc/client.rs

```diff
@@ -61,15 +61,20 @@ pub struct Client {
 }
 
 impl Client {
-    pub(crate) fn new(rpc_addr: &str) -> Self {
+    pub(crate) fn new(rpc_addr: &str) -> Result<Self> {
         let connector = JsonRpcClient::new_client();
-        let rpc_client = connector.connect(rpc_addr);
+        let mut rpc_client = connector.connect(rpc_addr);
+        if let Ok(api_key) = std::env::var("NEAR_RPC_API_KEY") {
```

Comment:
I find it quite a bit ugly to configure anything in library via env variables. What if I need to connect to two different APIs (e.g. testnet and mainnet) that have different API keys? I feel that `Client` type should just have a separate method to configure API key for `self.rpc_client`, instead of trying to configure it in the `new` method while trying to sneak it in via env variable.